### PR TITLE
groups: connectivity checks

### PR DESCRIPTION
**Pull request by jamesacklin**
Adds stoplight-style connection status (with title attributes) to:

- 1:1 DM headers in Talk
- ChatInfo modals for multi-DMs in Talk
- Profile modals in Groups
- Group host search in Discover / Groups
- Channel headers in Groups

Hides the connection status in the Group Admin view and the Profile for your own ship; displays a green-light status in channel headers and the ChatInfo modal.

Fixes LAND-664
Fixes LAND-391

### CODE REVIEW

**Comment by jamesacklin** [comment]
@urcades @dnbrwstr Not opposed to more contrast in the forms of our status indicators here, since we're differentiating mostly with color in the more space-constrained applications (hiding the label text).

The three major states are:
- Checking
- Unable to reach
- Connected

Not a directive, but a question/food for thought.

File: ui/src/components/ShipConnection.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+interface ShipConnectionProps {
+  ship: string;
+  status?: ConnectionStatus;
+  showBullet?: boolean;
+  showText?: boolean;
+  className?: string;
+}
+
+function getCompletedText(status: ConnectionCompleteStatus, ship: string) {
```
Comment (patosullivan) [inline ui/src/components/ShipConnection.tsx:18]: We may want to move these functions into utils

File: ui/src/state/vitals.ts [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+  useEffect(() => {
+    api.subscribe({
+      app: 'vitals',
+      path: `/status/${ship}`,
+      event: (data: ConnectionUpdate) => {
+        queryClient.setQueryData(['vitals', ship], data);
+      },
+    });
+    setSubbed(true);
+  }, [ship]);
```
Comment (patosullivan) [inline ui/src/state/vitals.ts:167]:
```suggestion
  }, [ship, queryClient]);
```

**Review by patosullivan — APPROVED** [review]
Two small things, otherwise lgtm

File: ui/src/components/ShipConnection.tsx (hunk 1 above)
Comment (jamesacklin) [reply ui/src/components/ShipConnection.tsx]: Handled, chef!

**Pull request merged**